internal/access/identity.go

```diff
@@ -143,21 +139,18 @@ func GetContextProviderIdentity(c *gin.Context) (*models.Provider, string, error
 }
 
 // UpdateIdentityInfoFromProvider calls the identity provider used to authenticate this user session to update their current information
-func UpdateIdentityInfoFromProvider(c *gin.Context, oidc providers.OIDCClient) error {
+func UpdateIdentityInfoFromProvider(c RequestContext, oidc providers.OIDCClient) error {
+	// does not need authorization check, this action is limited to the calling user
 	ctx := c.Request.Context()
 
 	// added by the authentication middleware
-	identity := AuthenticatedIdentity(c)
+	identity := c.Authenticated.User
 	if identity == nil {
 		return errors.New("user does not have session with an identity provider")
 	}
 
-	// does not need authorization check, this action is limited to the calling user
-	db := getDB(c)
-
-	accessKey := currentAccessKey(c)
-
-	provider, err := data.GetProvider(db, data.ByID(accessKey.ProviderID))
+	db := c.DBTxn
```

Comment:
c.DB or c.Tx might make this easier to use everywhere without setting it to a variable.

Reply:
long term I'd like to just pass this ctx down to the data layer.

Reply:
I only used a variable here to reduce the diff. Generally I would expect most code to use `c.DBTxn` directly. We could shorten it to something like `c.Tx`.

I would also use to pass something like this down to the data layer, but I also don't want the data layer to depend on things like `http.Request`, and other properties we might store on this.

So what I'm thinking is that the data layer functions will accept some kind of interface, probably `ReadTxn` and `WriteTxn`, and that interface will be implemented by either this `RequestContext`, or the request context will provide a method that returns the implementation.

I think that should allow the data layer access to things like the authenticated user, and org, without it having to depend on the `RequestContext` directly.